src/neo4j_graphrag/experimental/pipeline/types.py

```diff
@@ -35,3 +37,7 @@ class ConnectionConfig(BaseModel):
 class PipelineConfig(BaseModel):
     components: list[ComponentConfig]
     connections: list[ConnectionConfig]
+
+
+EntityInputType = Union[str, dict[str, Any]]
+RelationInputType = Union[str, dict[str, Any]]
```

Comment:
Is there a way to avoid using `Any` here?

Reply:
I've tried to make Pydantic output the type based on the `BaseModel`, but I failed. 

The real type would be something like:
```
dict[str, Union[str, list[dict[str, str]]]
```

Assuming (key => type):

- label => str
- description => str
- properties => list[]
  - name: str
  - description: str
  - type: str 

Do you still prefer the "real" type?

Reply:
Imo I think we should have that here otherwise it defeats the purpose of having Pydantic. Are there any issues with having this?

Reply:
No, actually I didn't do it because I thought it would be much more complicated, but it's still readable. Will update with a comment to explain where it comes from.

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/176/commits/f2f467ff9f80c73d482f9162b18ea35278a45efc